src/main/java/uk/gov/hmcts/ccd/data/user/AttributeBasedJurisdictionsResolver.java

```diff
@@ -3,33 +3,71 @@
     private final RoleAssignmentService roleAssignmentService;
+    private final CaseDefinitionRepository caseDefinitionRepository;
 
     public AttributeBasedJurisdictionsResolver(@Qualifier(CachedUserRepository.QUALIFIER)
-                                                     UserRepository userRepository,
-                                               RoleAssignmentService roleAssignmentService) {
+                                                   UserRepository userRepository,
+                                               RoleAssignmentService roleAssignmentService,
+                                               @Qualifier(CachedCaseDefinitionRepository.QUALIFIER)
+                                                   CaseDefinitionRepository caseDefinitionRepository) {
         this.userRepository = userRepository;
         this.roleAssignmentService = roleAssignmentService;
+        this.caseDefinitionRepository = caseDefinitionRepository;
     }
 
     @Override
     public List<String> getJurisdictions() {
         String userId = this.userRepository.getUser().getId();
         RoleAssignments roleAssignments = roleAssignmentService.getRoleAssignments(userId);
+        return filterJurisdictions(roleAssignments);
+    }
+
+    private List<String> filterJurisdictions(RoleAssignments roleAssignmentsList) {
+        List<String> jurisdictions = new ArrayList<>();
+        roleAssignmentsList.getRoleAssignments().stream()
+            .filter(roleAssignment -> !roleAssignment.getGrantType().equals(GrantType.BASIC.name()))
+            .forEach(roleAssignments -> {
+                if (Objects.nonNull(roleAssignments.getAttributes().getJurisdiction())) {
+                    doesJurisdictionExist(roleAssignments
+                        .getAttributes().getJurisdiction().get(), jurisdictions);
+                } else if (Objects.nonNull(roleAssignments.getAttributes().getCaseType())) {
+                    doesJurisdictionExist(caseDefinitionRepository
+                        .getCaseType(roleAssignments
+                            .getAttributes().getCaseType().get()).getJurisdictionId(), jurisdictions);
+                } else if (Objects.nonNull(roleAssignments.getRoleType())
+                    && roleAssignments.getRoleType().equalsIgnoreCase(ORGANISATION)) {
+                    caseDefinitionRepository.getAllJurisdictionsFromDefinitionStore()
+                        .forEach(jurisdictionDefinition ->
+                            doesJurisdictionExist(jurisdictionDefinition.getId(), jurisdictions));
+                }
+            });
+        return jurisdictions;
+    }
 
-        return roleAssignments.getJurisdictions();
+    private List<String> doesJurisdictionExist(String jurisdiction, List<String> jurisdictions) {
+        if (!jurisdictions.contains(jurisdiction.toLowerCase())) {
+            jurisdictions.add(jurisdiction.toLowerCase());
+        }
+        return jurisdictions;
```

Comment:
Do we need this method at all when we already filter `distinct` in `DefaultJurisdictionResolver`? For efficiency, breaking the loop whenever it hits the final `else if` would still probably be best

Reply:
done